src/nvim/drawline.c

```diff
@@ -1067,12 +1065,15 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
   buf_T *buf = wp->w_buffer;
   bool end_fill = (lnum == buf->b_ml.ml_line_count + 1);
 
+  const bool has_fold = foldinfo.fi_level != 0 && foldinfo.fi_lines > 0;
+  const bool has_foldtext = has_fold && strlen(wp->w_p_fdt) != 0;
```

Comment:
I think we usually use this to check whether an option is empty?
```suggestion
  const bool has_foldtext = has_fold && *wp->w_p_fdt != NUL;
```